### ISSUE TITLE
Mentions from slack to irc doesn't show names

### ISSUE THREAD

**Issue by cristeahub** [ekmartin/slack-irc#9]
![image](https://cloud.githubusercontent.com/assets/1488921/6554014/2383a168-c657-11e4-8fbd-a6a46fd28831.png)
This makes it hard to follow a conversations from irc.

**Comment by ekmartin** [comment]
Sadly this isn't possible with the current outgoing Slack API, see https://twitter.com/SlackAPI/status/575338812806582272 ([question](https://twitter.com/ekmartinb/status/574892570909765632)) :pensive:.

**Comment by captn3m0** [comment]
Yay! I can help with this.

I've worked on this problem a lot of times (in my many slack-integrations), and my solution has been integrated into slack-utils (on npm) now.

But yeah, it will make the setup complex by requiring an API key. Once you have the key, you can just call the [parseMessage method](https://github.com/captn3m0/slack-utils/blob/master/api.coffee#L66), which will parse all mentions of channels, usernames and more.

If this sounds fine to you, I might send a PR.

**Comment by ekmartin** [comment]
It could work if providing an API key is optional, so that it only retrieves channels and users and parses messages if an API key is given.

Feel free to give it a go if you want to :)

**Comment by ekmartin** [comment]
Actually, I think it might be better to use Slack's RTM API and their bot API (http://slackhq.com/post/104688116560/rtm-api), probably using https://github.com/slackhq/node-slack-client. 

Think it'll be better all around, and would also solve this issue, I'll give it a try.

**Comment by captn3m0** [comment]
Using bots would replace the webhooks. However, parsing the userids to usernames would still require the web API.

**Comment by ekmartin** [comment]
The `rtm.start` API-call returns a whole load of info, including a list of all users/channels/groups: https://api.slack.com/methods/rtm.start

**Comment by ekmartin** [comment]
@captn3m0 - you should send a pull request to https://github.com/slackhq/node-slack-client with your `parseMessage`-function, could probably be implemented here: https://github.com/slackhq/node-slack-client/blob/master/src%2Fmessage.coffee
